azure/azuredb/azuredb.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package azuredb contains Wire providers that are common across Azure Database.
+package azuredb
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// A CertPoolProvider returns a certificate pool that contains the Azure CA certificate.
+type CertPoolProvider interface {
+	GetCertPool(context.Context) (*x509.CertPool, error)
+}
+
+// AzureCertFetcher is a default CertPoolProvider that can fetch CA certificates from
+// any publicly accessible URI or File.
+type AzureCertFetcher struct {
+	Client *http.Client
+	// certLocation can be a remote endpoint or a file path
+	certLocation string
+	useHTTP      bool
+}
+
+// NewAzureCertFetcher constructs a new *AzureCertFetcher.
+// See https://docs.microsoft.com/en-us/azure/mysql/howto-configure-ssl.
+func NewAzureCertFetcher(caBundleLocation string) (*AzureCertFetcher, error) {
```

Comment:
For easier use in Wire, it would be good to have a version of this function that uses the default remote URL.

Reply:
Make sense, new constructor with default added.
`azuredb.NewAzureCertFetcherWithDefault()`